src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -216,27 +197,34 @@ public TypeProvider FindTypeForSchema(Schema schema)
             return result;
         }
 
-        private void BuildModels()
-        {
-            _models = new Dictionary<Schema, TypeProvider>();
-            _resourceModels = new Dictionary<Schema, TypeProvider>();
 
-            var allSchemas = _codeModel.Schemas.Choices.Cast<Schema>()
-                .Concat(_codeModel.Schemas.SealedChoices)
-                .Concat(_codeModel.Schemas.Objects)
-                .Concat(_codeModel.Schemas.Groups);
+        private Dictionary<Schema, TypeProvider> BuildModels()
+        {
+            var models = new Dictionary<Schema, TypeProvider>();
 
-            foreach (var schema in allSchemas)
+            foreach (var schema in _allSchemas)
             {
                 /*if (_context.Configuration.AzureArm && _operationGroups.ContainsKey(schema.Name))
                 {
-                    _resourceModels.Add(schema, BuildModel(schema));
+                    continue;
+                }*/
+                models.Add(schema, BuildModel(schema));
+            }
+            return models;
+        }
+
+        private Dictionary<Schema, TypeProvider> BuildResourceModels()
+        {
+            var resourceModels = new Dictionary<Schema, TypeProvider>();
+
+            foreach (var schema in _allSchemas)
```

Comment:
skip this for loop if _context.Configuration.AzureArm is false since it will result in an empty list anyways